.github/workflows/publish.yml

```
@@ -0,0 +1,49 @@
+name: Publish to PyPI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
```

We probably don't want this to trigger for every push to `main` or any tag that is pushed. Otherwise it will just try to upload with each build, and since the version isn't updated it will just fail.

We could have this action trigger every time we push a version specific tag, so when we bump a version and are ready to release, then we publish. We should be more specific about the tags though, and only trigger when it's a valid version number.
```suggestion
on:
  push:
    tags:
      - '[0-9]+.[0-9]+.[0-9]+'
```
That way we can have other tags that don't trigger this action if we want.